src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -77,50 +78,104 @@ class WorkspaceNewRoomPage extends React.Component {
     }
 
     /**
-     * Called when a workspace is selected. Also calls checkAndModifyRoomName,
-     * which displays an error if the given roomName exists on the newly selected workspace.
-     * @param {String} policyID
+     * Called when the "Create Room" button is pressed.
      */
-    onWorkspaceSelect(policyID) {
-        this.setState({policyID});
-        this.checkAndModifyRoomName(this.state.roomName);
+    onSubmit() {
+        if (!this.validate()) {
+            return;
+        }
+        Report.createPolicyRoom(
+            this.state.policyID,
+            this.state.roomName,
+            this.state.visibility,
+        );
     }
 
     /**
-     * Called when the "Create Room" button is pressed.
+     * Called when the onSubmit() method is triggered.
+     * which validates the form fields.
```

Comment:
```suggestion
     * Validates the form fields and store errors in the state.
     * Returns false when validation fails.
```

Reply:
OK. I see. I will revisit the style guide for a refresher. We can remove the comment then.

Reply:
Okay I will remove the comment then like this:
```
    /**
     * @returns {Boolean}
     */
```